data_collection/gazette/spiders/ap_macapa.py

```diff
@@ -10,60 +12,65 @@ class ApMacapaSpider(BaseGazetteSpider):
+        end_date = self.end_date.strftime("%d/%m/%Y") if self.end_date else ""
+        self.logger.debug(f"Start Date: {start_date} End Date: {end_date}")
+        params = {
             "s": "",
             "post_type": "official_diaries",
             "search": "official_diaries",
-            "official_diary_number": "",
+            "official_diary_initial_date": start_date,
+            "official_diary_final_date": end_date,
         }
-        while target_date <= self.end_date:
-            formatted_date = target_date.strftime("%d/%m/%Y")
-            data.update(
-                {
-                    "official_diary_initial_date": formatted_date,
-                    "official_diary_final_date": formatted_date,
-                }
-            )
+        yield scrapy.FormRequest(
+            url=base_url,
+            method="GET",
+            formdata=params,
+            callback=self._pagination_requests,
+            cb_kwargs={"params": params},
+        )
 
-            yield scrapy.FormRequest(
-                url=base_url, formdata=data, method="GET", meta={"date": target_date}
-            )
-            target_date = target_date + dt.timedelta(days=1)
+    def _pagination_requests(self, response, params):
+        pages = response.xpath("//a[@class='page-numbers']/text()").getall()
+        last_page = int(pages[-1]) if pages else None
+        if last_page:
+            for next_page in range(1, last_page + 1):
+                self.logger.debug(f"Page {next_page} of {last_page}")
+                next_page_url = "http://macapa.ap.gov.br/page/{page}/".format(
+                    **{"page": next_page}
+                )
+                yield scrapy.FormRequest(
+                    url=next_page_url,
+                    method="GET",
+                    formdata=params,
+                    callback=self.parse,
+                )
+        else:
+            self.logger.debug("One page only")
+            yield from self.parse(response)
 
     def parse(self, response):
         # Extract Items
-        links = response.xpath('//i[@class="fa fa-file-pdf-o"]/parent::a')
-        links = links.xpath("@href").getall()
-
-        gazette_date = response.meta["date"]
-
-        if len(links) == 0:
-            self.logger.warning(
-                "No gazettes found for date {date}".format(date=gazette_date)
-            )
-
-        for index, file_url in enumerate(links):
+        extract_number_date = re.compile(
+            r".+?(?P<num>\d{4}).+?(?P<date>\d\d/\d\d/\d\d\d\d)"
+        )
+        divs = response.xpath("//div[@class='panel-body']")[1:]
+        for div in divs:
+            url = div.xpath("./a/@href").get()
+            text = div.xpath("./a/h4/text()").get()
+            num_date = re.match(extract_number_date, text)
+            edition_number = num_date.group("num")
+            date = num_date.group("date")
+            date = dateparser.parse(date, languages=["pt"]).date()
```

Comment:
**sugestão** Se o formato da data é sempre o mesmo padrão, não há necessidade de usar a `dateparser` (que é custosa para executar). O built-in `strptime` do Python é suficiente.

 ```
gazette_date = datetime.datetime.strptime(gazette_raw_date, "%d/%m/%Y").date()
```